### PR TITLE
VY-203 Trim form data before saving

### PR DESCRIPTION
**Pull request by lf-kiranpariyar**
@lf-romitamgai . Please review the code.

### CODE REVIEW

**Comment by bhuwang** [comment]
Can one of the admins verify this patch?

File: frontend/src/js/components/blood-group/BloodGroupForm.js [hunk 1]
```diff
@@ -60,12 +62,18 @@ class BloodGroupForm extends React.Component {
...
     if (this.formValidator.isValid(requiredField)) {
-      this.props.actions.submitForm(resourceConstants.BLOOD_GROUPS, this.props.selectedItem.bloodGroups, this.props.params.id);
+      let bloodGroup = ObjectUtil.trimObject(this.getFormData());
+      this.props.actions.submitForm(resourceConstants.BLOOD_GROUPS, bloodGroup, this.props.params.id);
     } else {
       Toastr.error(messageConstants.FORM_INVALID_SUBMISSION_MESSAGE, messageConstants.TOASTR_ERROR);
     }
   }
 
+  getFormData() {
```
Comment (lf-romitamgai) [inline frontend/src/js/components/blood-group/BloodGroupForm.js]: you don't need this method, directly clone the object in calling method

File: frontend/src/js/components/designation/DesignationForm.js [hunk 2]
```diff
@@ -60,12 +62,18 @@ class DesignationForm extends React.Component {
...
     if (this.formValidator.isValid(requiredField)) {
-      this.props.actions.submitForm(resourceConstants.DESIGNATIONS, this.props.selectedItem.designations, this.props.params.id);
+      let designation = ObjectUtil.trimObject(this.getFormData());
+      this.props.actions.submitForm(resourceConstants.DESIGNATIONS, designation, this.props.params.id);
     } else {
       Toastr.error(messageConstants.FORM_INVALID_SUBMISSION_MESSAGE, messageConstants.TOASTR_ERROR);
     }
   }
 
+  getFormData() {
```
Comment (lf-romitamgai) [inline frontend/src/js/components/designation/DesignationForm.js]: same for here

File: frontend/src/js/components/division/DivisionForm.js [hunk 3]
```diff
@@ -60,12 +62,18 @@ class DivisionForm extends React.Component {
...
     if (this.formValidator.isValid(requiredField)) {
-      this.props.actions.submitForm(resourceConstants.DIVISIONS, this.props.selectedItem.divisions, this.props.params.id);
+      let division = ObjectUtil.trimObject(this.getFormData());
+      this.props.actions.submitForm(resourceConstants.DIVISIONS, division, this.props.params.id);
     } else {
       Toastr.error(messageConstants.FORM_INVALID_SUBMISSION_MESSAGE, messageConstants.TOASTR_ERROR);
     }
   }
 
+  getFormData() {
```
Comment (lf-romitamgai) [inline frontend/src/js/components/division/DivisionForm.js]: change this in all crud

File: frontend/src/js/components/employee/employee-form/EmployeeForm.js [hunk 4]
```diff
@@ -202,8 +206,13 @@ class EmployeeForm extends React.Component {
   }
 
   saveEmployee() {
-    let employee = this.getFormData();
-
+    let { employee, phones, socialAccounts, vehicles } = ObjectUtil.trimObject(this.getFormData());
```
Comment (lf-romitamgai) [inline frontend/src/js/components/employee/employee-form/EmployeeForm.js]: nice

File: frontend/src/js/components/employee/employee-form/tabbed-components/HRInfo.js [hunk 5]
```diff
@@ -141,13 +131,13 @@ class HRInfo extends React.Component {
...
+              <Select
                 name="divisions"
-                value={this.props.employee && this.mapDivisionsToLabelAndValue(this.props.employee.divisions)}
                 ref="divisions"
-                loadOptions={this.getSelectOptions}
-                onChange={this.props.handleMultiSelectChange}
                 placeholder="Please Select"
+                onChange={this.props.handleMultiSelectChange}
+                options={this.getSelectOptions(this.props.divisions)}
+                value={this.props.employee && this.mapDivisionsToLabelAndValue(this.props.employee.divisions)}
```
Comment (lf-romitamgai) [inline frontend/src/js/components/employee/employee-form/tabbed-components/HRInfo.js]: also check for this.props.employee.divisions before calling the method

**Pull request merged**